packages/react-i18n/src/i18n.ts

```diff
@@ -157,9 +162,17 @@ export default class I18n {
       );
     }
 
+    const {style = undefined, ...formatOptions} = options || {};
+
+    if (style) {
+      return style === DateStyle.Humanize
+        ? this.humanizeDate(date, formatOptions)
+        : this.formatDate(date, {...formatOptions, ...dateStyle[style]});
```

Comment:
should the options expansion be in the other order here?

Reply:
Hmmm, in this way, we can't overwrite dateStyle keys, but at the same time, if you pass a key, you want it to follow our set of styles. So I'd say it's good this way. 🤔